metadata/interface.go

```diff
@@ -54,19 +55,18 @@ type Manager interface {
 	// mapping of parameters of the form:
 	// componentName => {param key => param val}
 	// i.e.: "nginx" => {"replicas" => 1, "name": "nginx"}
-	GetEnvironmentParams(name string) (map[string]param.Params, error)
+	GetEnvironmentParams(name, nsName string) (map[string]param.Params, error)
 	SetEnvironmentParams(env, component string, params param.Params) error
 
 	// Environment API.
 	CreateEnvironment(name, uri, namespace, spec string) error
 	DeleteEnvironment(name string) error
-	GetEnvironments() (app.EnvironmentSpecs, error)
-	GetEnvironment(name string) (*app.EnvironmentSpec, error)
+	GetEnvironments() (map[string]env.Env, error)
+	GetEnvironment(name string) (*env.Env, error)
 	SetEnvironment(name, desiredName string) error
 
-	// Spec API.
-	AppSpec() (*app.Spec, error)
-	WriteAppSpec(*app.Spec) error
+	// Destination API.
```

Comment:
I would just bundle this with the `// Environment API` group since destinations are a part of environments.